### PR TITLE
SLS-492 Add transaction id support for deltas

### PR DESCRIPTION
**Pull request by quchenhao**
Delta can be read back by the same node so it also needs to handle transaction id.

### CODE REVIEW

File: src/include/cell_inline.h [hunk 1]
```diff
@@ -1260,6 +1329,11 @@ __wt_cell_unpack_delta(WT_SESSION_IMPL *session, WT_DELTA_CELL *cell, WT_CELL_UN
     } else {
         WT_TIME_WINDOW_INIT(&unpack->tw);
 
+        if (F_ISSET(unpack, WT_DELTA_HAS_START_TXN_ID)) {
+            ret = __wt_vunpack_uint(&p, 0, &unpack->tw.start_txn);
+            WT_ASSERT(session, ret == 0);
```
Comment (korteland) [inline src/include/cell_inline.h:1334]: I realise it's not something new in this PR, but this unpacking code that relies on assertions for error-checking make me nervous. Is it reasonable to have them as an `ASSERT_ALWAYS`, at lease?

**Review by korteland — COMMENTED** [review]
I don't understand the existing visibility code well enough to be really confident in my review, but it looks fine as far as I can tell.

One question, is it worth adding tests to this? I realise the existing visibility tests will catch this once they're running with disagg, but in the meantime another `test_oligarch` or something would provide some assurance that this is working properly.

File: src/include/cell_inline.h (hunk 1 above)
Comment (quchenhao) [reply src/include/cell_inline.h:1334]: This is perf critical code. So in the cell unpacking code, we only use WT_ASSERT.

Comment (korteland) [reply src/include/cell_inline.h:1334]: OK. Thanks for explaining it.

**Review by korteland — APPROVED** [review]
LGTM

**Pull request merged**